### PR TITLE
web: compute update status on the server side instead of the client side.

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc, @landism,

Please review the following commits I made in branch nicks/ch10662:

c46c3c3fcb81fe414a2bcba662c04239bffb92fa (2021-01-07 18:00:50 -0500)
web: compute update status on the server side instead of the client side.
fixes https://github.com/tilt-dev/tilt/issues/4041
fixes https://github.com/tilt-dev/tilt/issues/4009

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/store/engine_state.go [hunk 1]
```diff
@@ -667,6 +667,32 @@ func (ms *ManifestState) HasPendingChangesBeforeOrEqual(highWaterMark time.Time)
 	return ok, earliest
 }
 
+func (ms *ManifestState) UpdateStatus(triggerMode model.TriggerMode) model.UpdateStatus {
```
Comment (nicks) [inline internal/store/engine_state.go:670]: the whole thrust of this PR is just taking this 'if' branch from the client and moving it to the server, and improving it a bit with stuff the server knows (like whether a local resource has an update command)

File: web/src/status.tsx [hunk 2]
```diff
@@ -1,33 +1,25 @@
...
+  } else if (
+    res.updateStatus == UpdateStatus.NotApplicable ||
+    res.updateStatus == UpdateStatus.None
+  ) {
     return ResourceStatus.None
+  } else if (buildWarnings(res).length > 0) {
+    return ResourceStatus.Warning
+  } else if (res.updateStatus == UpdateStatus.Ok) {
+    return ResourceStatus.Healthy
+  } else if (res.updateStatus == UpdateStatus.Error) {
```
Comment (maiamcc) [inline web/src/status.tsx:19]: seems like we should show error above warning, doesn't this want to be above ln 15?

**Review by maiamcc — APPROVED** [review]
💯

File: web/src/status.tsx (hunk 2 above)
Comment (nicks) [reply web/src/status.tsx]: oops, good catch! fixed

**Pull request merged**